From 0deb63dd394cbdf2e6b75f57b53c0ba6091d2d40 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Sat, 15 Jun 2019 12:58:30 +0200
Subject: [PATCH 1/1] efi_loader: SetMode() parameters check

If EFI_GRAPHICS_OUTPUT_PROTOCOL.SetMode() is called with an invalid mode,
return EFI_UNSUPPORTED.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_loader/efi_gop.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/lib/efi_loader/efi_gop.c b/lib/efi_loader/efi_gop.c
index c44a1cfc47..676463f2de 100644
--- a/lib/efi_loader/efi_gop.c
+++ b/lib/efi_loader/efi_gop.c
@@ -60,12 +60,14 @@ out:
 
 static efi_status_t EFIAPI gop_set_mode(struct efi_gop *this, u32 mode_number)
 {
+	efi_status_t ret = EFI_SUCCESS;
+
 	EFI_ENTRY("%p, %x", this, mode_number);
 
-	if (mode_number != 0)
-		return EFI_EXIT(EFI_INVALID_PARAMETER);
+	if (mode_number)
+		ret = EFI_UNSUPPORTED;
 
-	return EFI_EXIT(EFI_SUCCESS);
+	return EFI_EXIT(ret);
 }
 
 static __always_inline struct efi_gop_pixel efi_vid16_to_blt_col(u16 vid)
-- 
2.20.1

